services/navigation/path.go

```diff
@@ -32,16 +42,26 @@ func (p *Path) GeoPoints() []*geo.Point {
 }
 
 // PathSliceToProto converts a slice of Path into an equivalent Protobuf message.
-func PathSliceToProto(paths []*Path) []*pb.Path {
+func PathSliceToProto(paths []*Path) ([]*pb.Path, error) {
+	if paths == nil {
+		return nil, errNilSlice
+	}
 	var pbPaths []*pb.Path
 	for _, path := range paths {
-		pbPaths = append(pbPaths, PathToProto(path))
+		pbPath, err := PathToProto(path)
+		if err != nil {
+			return nil, err
+		}
+		pbPaths = append(pbPaths, pbPath)
 	}
-	return pbPaths
+	return pbPaths, nil
 }
 
 // PathToProto converts the Path struct into an equivalent Protobuf message.
-func PathToProto(path *Path) *pb.Path {
+func PathToProto(path *Path) (*pb.Path, error) {
+	if path == nil {
+		return nil, errNilPath
+	}
 	var pbGeoPoints []*commonpb.GeoPoint
 	for _, pt := range path.geoPoints {
```

Comment:
I think this will panic if `pt` is nil

Reply:
I agree it'll panic, but at the same time it is impossible to create a path struct with no geoPoints, so I think this should be safe.